### PR TITLE
Upgrade phpstan to level 6

### PR DESCRIPTION
**Pull request by mvorisek**
rebase after https://github.com/atk4/core/pull/317

### CODE REVIEW

File: docs/container.rst [hunk 1]
```diff
@@ -121,7 +121,7 @@ Methods
 
     Return object if it exits in collection and false otherwise
 
-.. php:method:: _getFromCollection(string $name, string $collection)o    
+.. php:method:: _getFromCollection(string $name, string $collection)o
```
Comment (georgehristov) [inline docs/container.rst]: I think this "o" should be removed?

**Review by georgehristov — APPROVED** [review]
LGTM

File: docs/container.rst (hunk 1 above)
Comment (mvorisek) [reply docs/container.rst]: I trimmed only trailing WS :), but yes

File: tests/CollectionTraitTest.php [hunk 2]
```diff
@@ -31,30 +33,33 @@ public function testBasic()
...
+        /** @var FieldMockCustom $surnameField */
+        $surnameField = $m->addField('surname', [FieldMockCustom::class]);
 
-        $this->assertSame('app', $surname->getApp()->name);
+        $this->assertSame('app', $surnameField->getApp()->name);
 
-        $this->assertSame('form-fields_surname', $surname->name);
-        $this->assertSame($surname->getOwner(), $m);
+        $this->assertSame('form-fields_surname', $surnameField->name);
+        $this->assertSame($m, $surnameField->getOwner());
```
Comment (mvorisek) [inline tests/CollectionTraitTest.php:53]: discussion: https://github.com/phpstan/phpstan/discussions/4985

**Pull request merged**